regress/pesterTests/KeyUtils.Tests.ps1

```diff
@@ -153,43 +188,157 @@ Describe "E2E scenarios for ssh key management" -Tags "Scenario" {
+            
+            #clean up
+            cmd /c "ssh-add -d $keyFilePath 2> nul "
         }
         
+        It "$tC.$tI-  ssh-add - negative (other account can access private key file)" {
+            #setup to have current user as owner and grant it full control        
+            Set-FileOwnerAndACL -FilePath $keyFilePath -Owner $currentUser -OwnerPerms "FullControl"         
+
+            #add ssouser to access the private key            
+            Add-PermissionToFileACL -FilePath $keyFilePath -User $objUser -Perm "Read"
+
+            cmd /c "ssh-add $keyFilePath < $nullFile 2> nul "
+            $LASTEXITCODE | Should Not Be 0
+
+            $allkeys = ssh-add -L
+            $pubkeyraw = ((Get-Content "$keyFilePath.pub").Split(' '))[1]            
+            ($allkeys | where { $_.contains($pubkeyraw) }).count | Should Be 0
+        }
+
+        It "$tC.$tI - ssh-add - negative (the private key has wrong owner)" {
+            #setup to have ssouser as owner and grant it full control
+            Set-FileOwnerAndACL -FilePath $keyFilePath -owner $objUser -OwnerPerms "Read, Write"
+            Add-PermissionToFileACL -FilePath $keyFilePath -User $adminsAccount -Perm "FullControl"
+
+            cmd /c "ssh-add $keyFilePath < $nullFile 2> nul "
+            $LASTEXITCODE | Should Not Be 0
+
+            $allkeys = ssh-add -L
+            $pubkeyraw = ((Get-Content "$keyFilePath.pub").Split(' '))[1]            
+            ($allkeys | where { $_.contains($pubkeyraw) }).count | Should Be 0
+        }
+
+        It "$tC.$tI - ssh-add- negative (the owner is denied Read perm on private key)" {
+            #setup to have local ssytem account as owner and grant it full control
+            Set-FileOwnerAndACL -FilePath $keyFilePath -owner $systemAccount -OwnerPerms "FullControl"
+            Add-PermissionToFileACL -FilePath $keyFilePath -User $adminsAccount -Perm "FullControl"
+            #deny owner 
+            Add-PermissionToFileACL -FilePath $keyFilePath -User $systemAccount -Perm "Read, Write" -AccessType Deny
+
+            cmd /c "ssh-add $keyFilePath < $nullFile 2> nul "
+            $LASTEXITCODE | Should Not Be 0
+
+            $allkeys = ssh-add -L
+            $pubkeyraw = ((Get-Content "$keyFilePath.pub").Split(' '))[1]            
+            ($allkeys | where { $_.contains($pubkeyraw) }).count | Should Be 0
+        }
     }
 		
     Context "$tC - ssh-keyscan test cases" {
-        BeforeAll {$tI=1}
+        BeforeAll {
+            $tI=1
+            Remove-item (join-path $testDir "$tC.$tI.out.txt") -force -ErrorAction Ignore
+        }
+        BeforeEach {
+            $outputFile = join-path $testDir "$tC.$tI.out.txt"
+        }
         AfterAll{$tC++}
 
 		It "$tC.$tI - ssh-keyscan with default arguments" {
-			iex "ssh-keyscan -p 22 github.com 2>&1 > out.txt"
-			'out.txt' | Should Contain 'github.com ssh-rsa.*' 			
+			cmd /c "ssh-keyscan -p 22 github.com 2>&1 > $outputFile"
```

Comment:
please change replace GitHub.com with 127.0.0.1 for all the ssh-keyscan testcases
			cmd /c "ssh-keyscan -p $port 127.0.0.1 2>&1 > outputFile"
			$outputFile  | Should Contain '.*ssh-rsa.*'